packages/core/src/storage.ts

```diff
@@ -18,34 +18,46 @@ export interface IClientStorage {
  */
 export class ClientStorage implements IClientStorage {
   private readonly storage: Storage;
+  private readonly prefix: string;
 
-  constructor(storage?: Storage) {
+  constructor(storage?: Storage, prefix?: string) {
     this.storage = storage ?? (typeof localStorage !== 'undefined' ? localStorage : new MemoryStorage());
+    this.prefix = prefix ?? '@medplum:';
+  }
+
+  private makeKey(key: string): string {
+    return this.prefix + key;
   }
 
   clear(): void {
-    this.storage.clear();
+    const keys = this.storage instanceof MemoryStorage ? this.storage.keys() : Object.keys(this.storage);
```

Comment:
While `Object.keys(this.storage)` works for localStorage, this is not safe for any `Storage` implementation. Might need to create a new interface that extends `Storage` and adds a key method.

Reply:
https://github.com/medplum/medplum/pull/3806/commits/94daee94c5d7c35cd17842d35ac52ede45312e81 is an attempt at solving this issue